File: Version2/GO/ReadCurrenciesRestApi_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you: hire me!
 * $ go test
 */
package GO

import (
	"reflect"
	"testing"
)

func TestReadCurrenciesRestApi(t *testing.T) {
	tables := []struct {
		site string
		want []string
	}{
		{"https://kriptonita.com.br", []string{"BRL", "USD"}},
		{"https://www.havan.com.br", []string{"BRL"}},
		{"https://www.purplesunrise.com", []string{"GBP"}},
		{"https://truckpartshq.com", []string{"USD"}},
	}

	for _, table := range tables {
		got := getCurrenciesWithREST(table.site)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %v but the return was this: %v", table.want, got)
		}
	}
}

File: Version2/PHP/ReadCurrenciesRestApiTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("ReadCurrenciesRestApi.php");

class ReadCurrenciesRestApiTest extends TestCase {

    public function testReadCurrenciesRestApi() {
        $obj = new ReadCurrenciesRestApi();
        //Test 1
        $site = "https://kriptonita.com.br";
        $want = ["BRL","USD"];
        $got = $obj->getCurrenciesWithREST( $site );
        $this->assertEquals($want, $got);
        //Test 2
        $site = "https://www.havan.com.br";
        $want = ["BRL"];
        $got = $obj->getCurrenciesWithREST( $site );
        $this->assertEquals($want, $got);
        //Test 3
        $site = "https://www.purplesunrise.com";
        $want = ["GBP"];
        $got = $obj->getCurrenciesWithREST( $site );
        $this->assertEquals($want, $got);
        //Test 4
        $site = "https://truckpartshq.com";
        $want = ["USD"];
        $got = $obj->getCurrenciesWithREST( $site );
        $this->assertEquals($want, $got);
    }

}
